### PR TITLE
optimize: replace jwt error code with custom errno

### PR DESCRIPTION
**Pull request by BaiZe1998**
#### What type of PR is this?

optimize

#### What this PR does / why we need it (en: English/zh: Chinese):

en: override the error code of the default jwt process by configuring the response content to inject a custom errno before the easy_note initializes the api service.
zh: 在 easy_note 初始化 api 服务前，通过配置响应内容，为其注入自定义的 errno，从而覆盖默认 jwt 流程的错误码。

### CODE REVIEW

File: bizdemo/easy_note/cmd/api/main.go [hunk 1]
```diff
@@ -17,8 +17,11 @@ package main
 
 import (
 	"context"
+	"net/http"
 	"time"
 
+	"github.com/cloudwego/kitex-examples/bizdemo/easy_note/pkg/errno"
```
Comment (li-jin-gou) [inline bizdemo/easy_note/cmd/api/main.go:23]: need to format import style

File: bizdemo/easy_note/cmd/api/main.go [hunk 2]
```diff
@@ -49,6 +52,19 @@ func main() {
 			}
 			return jwt.MapClaims{}
 		},
+		LoginResponse: func(ctx context.Context, c *app.RequestContext, code int, token string, expire time.Time) {
```
Comment (li-jin-gou) [inline bizdemo/easy_note/cmd/api/main.go:55]: add some Handling Logic to return self-customed code response
- method not found 
- 404 not found
- panic and recovery

**Review by li-jin-gou — CHANGES_REQUESTED** [review]

File: bizdemo/easy_note/cmd/api/main.go (hunk 1 above)
Comment (BaiZe1998) [reply bizdemo/easy_note/cmd/api/main.go:23]: i got it, thx

File: bizdemo/easy_note/cmd/api/main.go (hunk 2 above)
Comment (BaiZe1998) [reply bizdemo/easy_note/cmd/api/main.go:55]: i got it, thx

File: bizdemo/easy_note/pkg/errno/errno.go [hunk 3]
```diff
@@ -21,12 +21,16 @@ import (
...
+	SuccessCode                = 0
+	ServiceErrCode             = 10001
+	ParamErrCode               = 10002
+	LoginErrCode               = 10003
+	UserNotExistErrCode        = 10004
+	UserAlreadyExistErrCode    = 10005
+	AuthorizationFailedErrCode = 10006
+	NoRouteErrCode             = 10007
+	NoMethodErrCode            = 10008
+	ServerPanicErrCode         = 10009
```
Comment (li-jin-gou) [inline bizdemo/easy_note/pkg/errno/errno.go:33]: ServerPanicErrCode seem equal to ServiceErrCode

File: bizdemo/easy_note/cmd/api/main.go [hunk 4]
```diff
@@ -66,6 +83,15 @@ func main() {
 		TimeFunc:      time.Now,
 	})
 
+	r.Use(recovery.Recovery(recovery.WithRecoveryHandler(
+		func(ctx context.Context, c *app.RequestContext, err interface{}, stack []byte) {
+			hlog.SystemLogger().CtxErrorf(ctx, "[Recovery] err=%v\nstack=%s", err, stack)
+			c.JSON(http.StatusInternalServerError, map[string]interface{}{
+				"code":    errno.ServerPanicErrCode,
+				"message": err.(string),
```
Comment (li-jin-gou) [inline bizdemo/easy_note/cmd/api/main.go:91]: i think it should contains stack and err.

**Review by li-jin-gou — CHANGES_REQUESTED** [review]

File: bizdemo/easy_note/README.md [hunk 5]
```diff
@@ -142,7 +142,7 @@ curl --location --request POST '127.0.0.1:8080/v1/user/login' \
 #### response
 ```javascript
 {
-    "code": 200,
+    "code": 0,
```
Comment (li-jin-gou) [inline bizdemo/easy_note/README.md:145]: Would it be better to improve the readme description and add other cases?

File: bizdemo/easy_note/pkg/errno/errno.go (hunk 3 above)
Comment (BaiZe1998) [reply bizdemo/easy_note/pkg/errno/errno.go:33]: i got it, thx

File: bizdemo/easy_note/cmd/api/main.go (hunk 4 above)
Comment (BaiZe1998) [reply bizdemo/easy_note/cmd/api/main.go:91]: i got it, thx

File: bizdemo/easy_note/README.md (hunk 5 above)
Comment (BaiZe1998) [reply bizdemo/easy_note/README.md:145]: I got it, thx

File: bizdemo/easy_note/README.md [hunk 6]
```diff
@@ -141,11 +188,18 @@ curl --location --request POST '127.0.0.1:8080/v1/user/login' \
...
+// successful
 {
-    "code": 200,
+    "code": 0,
     "expire": "2022-01-19T01:56:46+08:00",
     "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NDI1Mjg2MDYsImlkIjoxLCJvcmlnX2lhdCI6MTY0MjUyNTAwNn0.k7Ah9G4Enap9YiDP_rKr5HSzF-fc3cIxwMZAGeOySqU"
 }
+// failed
+{
+    "code": 10004,
```
Comment (li-jin-gou) [inline bizdemo/easy_note/README.md:199]: why not use 10006 instead of 1004?

File: bizdemo/easy_note/cmd/api/main.go [hunk 7]
```diff
@@ -77,5 +112,19 @@ func main() {
...
 	note1.DELETE("/:note_id", handlers.DeleteNote)
+
+	r.NoRoute(func(ctx context.Context, c *app.RequestContext) {
+		c.JSON(http.StatusNotFound, map[string]interface{}{
+			"code":    errno.NoRouteErrCode,
+			"message": errno.NoRouteErr.ErrMsg,
+			"data":    nil,
+		})
+	})
+	r.NoMethod(func(ctx context.Context, c *app.RequestContext) {
```
Comment (li-jin-gou) [inline bizdemo/easy_note/cmd/api/main.go:123]: refer to https://www.cloudwego.io/zh/docs/hertz/tutorials/basic-feature/route/#noroute-%E4%B8%8E-nomethod-%E4%BD%BF%E7%94%A8

**Review by li-jin-gou — CHANGES_REQUESTED** [review]

File: bizdemo/easy_note/README.md (hunk 6 above)
Comment (BaiZe1998) [reply bizdemo/easy_note/README.md:199]: i got it, thx

File: bizdemo/easy_note/cmd/api/main.go (hunk 7 above)
Comment (BaiZe1998) [reply bizdemo/easy_note/cmd/api/main.go:123]: i got it, thx

**Review by li-jin-gou — APPROVED** [review]

**Comment by li-jin-gou** [comment]
LGTM

**Pull request merged**